packages/iov-bns/src/bnsconnection.ts

```diff
@@ -416,6 +428,26 @@ export class BnsConnection implements BcpAtomicSwapConnection {
     );
   }
 
+  public async getUsernames(query: BnsUsernamesQuery): Promise<ReadonlyArray<BnsUsernameNft>> {
```

Comment:
very nice function, i like this.
good use of the paths and the arguments.

you are entering weave mastery :)

Reply:
:)